CONTRIBUTING.rst

```diff
@@ -0,0 +1,109 @@
+coverage results for only that module without running all of the DWI_ML
+tests. For example, if you are adding code to ``dwi_ml/core/utils.py``,
+you can use::
+
+    coverage run --source=dwi_ml.core.utils -m pytest -s --doctest-modules --verbose dwi_ml/core/tests/test_utils.py
+
+You can then use ``coverage report`` to view the results, or use
+``coverage html`` and open ``htmlcov/index.html`` in your browser for a nicely
+formatted interactive coverage report.
+
+Contributions to tests that extend test coverage in older modules that are not
+fully covered are very welcome !
+
+Code style
+----------
+
+Code contributions should be formatted according to the `DWI_ML Coding Style Guideline <./doc/devel/coding_style_guideline.rst>`_
+Please, read the document to conform your code contributions to the DWI_ML
+standard.
+
+Documentation
+-------------
+
+DWI_ML uses `Sphinx`_ to generate documentation. The `DWI_ML Coding Style Guideline <./doc/devel/coding_style_guideline.rst>`_
+contains details about documenting the contributions.
+
+Commit Messages
+===============
+
+Write your commit messages using the standard prefixes for DWI_ML commit
+messages:
+
+* ``BUG:`` Fix for runtime crash or incorrect result
+* ``COMP:`` Compiler error or warning fix
+* ``DOC:`` Documentation change
+* ``ENH:`` New functionality
+* ``PERF:`` Performance improvement
+* ``STYLE:`` No logic impact (indentation, comments)
+* ``WIP:`` Work In Progress not ready for merge
+
+The body of the message should clearly describe the motivation of the commit
+(**what**, **why**, and **how**). In order to ease the task of reviewing
+commits, the message body should follow the following guidelines:
+
+1. Leave a blank line between the subject and the body. This helps ``git log``
+   and ``git rebase`` work nicely, and allows to smooth generation of release
+   notes.
+2. Try to keep the subject line below 72 characters, ideally 50.
+3. Capitalize the subject line.
+4. Do not end the subject line with a period.
+5. Use the imperative mood in the subject line (e.g. ``BUG: Fix spacing not
+   being considered.``).
+6. Wrap the body at 80 characters.
+7. Use semantic line feeds to separate different ideas, which improves the
+   readability.
+8. Be concise, but honor the change: if significant alternative solutions were
+   available, explain why they were discarded.
+9. If the commit fixes a regression test, provide the link. If it fixes a
+   compiler error, provide a minimal verbatim message of the compiler error. If
+   the commit closes an issue, use the `GitHub issue closing keywords <https://help.github.com/en/articles/closing-issues-using-keywords>`_.
+
+Keep in mind that the significant time is invested in reviewing commits and
+pull requests, so following these guidelines will greatly help the people doing
+reviews.
+
+These guidelines are largely inspired by Chris Beam's `How to Write a Commit Message <https://chris.beams.io/posts/git-commit/>`_
+post.
```

Comment:
I see that it was also taken from ITK. Should it be cited?

Reply:
I was the one that added that to ITK with the contributions of other folks, but yeah, why not.

Reply:
Done in ad7ee3c.